File: 3. Linked List/4_get_node_value.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None

def get_node_value(head, index):
    current = head
    while current is not None:
        if index == 0:
            return current.val
        current = current.next
        index -= 1
    return None

File: 3. Linked List/4_get_node_value/4_get_node_value.java
class Node<T> {
    T val;
    Node<T> next;

    public Node(T val) {
        this.val = val;
        this.next = null;
    }
}

class Source {
    public static <T> T getNodeValue(Node<T> head, int index) {

        Node<T> current = head;

        while (current != null) {
            if (index == 0) {
                return current.val;
            }
            current = current.next;
            index--;
        }

        return null;
    }

}